src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -145,8 +145,14 @@ private static bool EnsureDirectory(string path)
 
         ~Repository()
         {
-            if (!Disposed)
+            try
+            {
                 Dispose(false);
+            }
+            catch (Exception ex) when (!Environment.HasShutdownStarted)
+            {
+                Console.Error.WriteLine($"Exception thrown in call to {GetType()}.Finalize:{Environment.NewLine}{ex}");
```

Comment:
@eerhardt had suggested that we shouldn't be writing to the console from libraries. Is there a better way we can log the errors? Maybe using the Channel.Trace or similar functions?

Reply:
If we log this message, the process is already in the process of crashing, which is arguably worse than logging from the library.

Reply:
📝 In the first iteration of this pull request, the new `catch` blocks did not rethrow the exception. I have since updated the pull request in two ways:

* Use `Console.WriteLine` instead of `Console.Error.WriteLine`. Only the former is captured by whatever strategy we are using for running tests in CI, which is where we really need to be able to see the messages.
* I added `throw;` statements to rethrow the exception. This means the pull request will not add stability in cases where these exceptions were already being thrown, but instead merely aims to give us data necessary to identify and resolve the original source of the exception.